Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -271,9 +243,55 @@ GameListCtrl::GameListCtrl(wxWindow* parent, const wxWindowID id, const wxPoint&
   Bind(wxEVT_MENU, &GameListCtrl::OnChangeDisc, this, IDM_LIST_CHANGE_DISC);
   Bind(wxEVT_MENU, &GameListCtrl::OnNetPlayHost, this, IDM_START_NETPLAY);
 
-  Bind(DOLPHIN_EVT_RELOAD_GAMELIST, &GameListCtrl::OnReloadGameList, this);
+  Bind(DOLPHIN_EVT_REFRESH_GAMELIST, &GameListCtrl::OnRefreshGameList, this);
+  Bind(DOLPHIN_EVT_RESCAN_GAMELIST, &GameListCtrl::OnRescanGameList, this);
 
   wxTheApp->Bind(DOLPHIN_EVT_LOCAL_INI_CHANGED, &GameListCtrl::OnLocalIniModified, this);
+
+  if (!disable_scanning)
+  {
+    m_scan_thread = std::thread([&] {
+      Common::SetCurrentThreadName("gamelist scanner");
+
+      if (SyncCacheFile(false))
+      {
+        // Account for changes outside the cache (just wii banners atm; could scan Ini here)
+        // Note this is on the initial load path. Further improvements could be made if updates
+        // from scan->ui threads described small changelists instead of updating the entire list
+        // at once.
+        bool cache_modified = false;
+        for (auto& file : m_cached_files)
```

Comment:
I don't suppose theres a standard/STL thing that expresses intent a little better (like, "check all and tell me if at least one of them returned true", eg. `std::any_of`/ `std::accumulate`/... except that all items are always checked instead of stopping for the first item that matches)?
Otherwise, some braces please, since this is two levels deep and the inside of the `for`-loop is more than one line.